src/CoreBluetooth/AdvertisementData.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Linq;
+
+using XamCore.ObjCRuntime;
+using XamCore.Foundation;
+using XamCore.CoreFoundation;
+
+namespace XamCore.CoreBluetooth {
+
+	public partial class AdvertisementData {
+		
+		public IDictionary <CBUUID, NSData> ServiceData {
+			get {
+				using (var nsDict = _ServiceData) {
```

Comment:
Yep, fixing generator sounds like the best option - and we might need that very soon anyway :)